client/redux/auth/authSlice.ts

```diff
@@ -0,0 +1,186 @@
+import {
+  createSlice,
+  PayloadAction,
+  createAsyncThunk,
+  ActionReducerMapBuilder,
+} from '@reduxjs/toolkit';
+import { HYDRATE } from 'next-redux-wrapper';
+
+import authService from './authService';
+import { catchError } from 'utils/handleAxiosError';
+import { SignInUpFormValues, User, AxiosResponseError } from 'shared/types';
+
+type InitialState = {
+  user: any;
+  isError: boolean;
```

Comment:
```suggestion
type InitialState = {
  user: SignInUpFormValues;
  isError: boolean;
```

Reply:
ahh for this one it should be like this

export type User = {
  user: {
    id: number;
    first_name: string;
    last_name: string;
    email: string;
    created_at: string;
    updated_at: string;
  };
  token: string;
};

export type InitialState = {
  user: User | null;
  isError: boolean;
  isSuccess: boolean;
  isLoading: boolean;
  error: AxiosResponseError;
};